improver/wxcode/modal_code.py

```diff
@@ -204,6 +209,11 @@ def process(self, cubes: CubeList) -> Cube:
             A single weather code cube with time bounds that span those of
             the input weather code cubes.
         """
+        # Set the record_run attribute on all cubes. This will survive the
+        # merge and be present on the output.
+        if self.record_run_attr is not None and self.model_id_attr is not None:
```

Comment:
If the user specifies the `record_run_attr` without the `model_id_attr`, it will be silently ignored. This isn't documented behaviour, and I think it should cause an Exception, probably in the `__init__` method.

Reply:
Exception added and coverage added to the tests.